examples/gno.land/p/demo/boardsv2/draft3/app.gno

```diff
@@ -0,0 +1,121 @@
+package boards
+
+import (
+	"gno.land/p/demo/boards/post"
+	"gno.land/p/demo/boards/post/plugin"
+	pluginreputation "gno.land/p/demo/boards/post/plugin/reputation"
+	plugintitle "gno.land/p/demo/boards/post/plugin/title"
+)
+
+const (
+	LevelBoard = iota
+	LevelThread
+	LevelComment
+)
+
+// App is the boards application.
+type App struct {
+	posts            post.Store
+	plugins          *plugin.Registry
+	maxCommentsDepth int
+	disableComments  bool
+	reputationPolicy pluginreputation.Policy
+}
+
+// New creates a new boards application.
+func New(st post.Store, options ...Option) App {
+	app := App{posts: st}
+	for _, apply := range options {
+		apply(&app)
+	}
+
+	app.plugins := plugin.NewRegistry(
+		plugintitle.New(st),   // Plugin for boards
+		plugintext.New(st),    // Plugin for text based threads
+		pluginpoll.New(st),    // Plugin for poll based threads
+		plugincomment.New(st), // Plugin for comments to the threads
+		pluginreputation.New(
+			pluginreputation.UsePolicy(app.reputationPolicy),
+			pluginreputation.AllowedPostLevels(post.LevelPost, post.LevelComment),
+		),
+	)
+	return app
+}
+
+func (a App) GetPost(path string) (_ *post.Post, found bool) {
+	return a.post.Get(path) // NOTE: Allowing this enables us to have generic functions like Lock
+}
+
+func (a App) GetBoard(path string) (_ Board, found bool) {
+	p, found := a.posts.Get(path)
```

Comment:
seems like similar logic was repeated in several functions we could have a getPost(level int) function

Reply:
Addressed in 9a60ee5